testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/core/basic/DisabledDefaultExceptionMapperTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2022 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.resteasy.test.core.basic;
+
+import java.net.URL;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.ApplicationPath;
+import jakarta.ws.rs.ConstrainedTo;
+import jakarta.ws.rs.RuntimeType;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.core.Application;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriInfo;
+import jakarta.ws.rs.ext.ExceptionMapper;
+import jakarta.ws.rs.ext.Provider;
+import jakarta.ws.rs.ext.Providers;
+
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.resteasy.setup.DisableDefaultExceptionMapperSetupTask;
+import org.jboss.resteasy.test.core.basic.resource.ExceptionResource;
+import org.jboss.resteasy.utils.TestUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests that the default {@link ExceptionMapper} is disabled.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@ServerSetup(DisableDefaultExceptionMapperSetupTask.class)
+public abstract class DisabledDefaultExceptionMapperTest {
+
+    @Inject
+    protected Providers providers;
+    @Inject
+    protected Client client;
```

Comment:
For my own education, I'm not sure where to look for the Client producer ...

Reply:
This was something I had added because there is talk about removing the `@Context` annotation. The bulk of them are created here https://github.com/resteasy/resteasy/blob/f8e03aabbca93fbaeecb21919343aa4484ae7ea1/resteasy-cdi/src/main/java/org/jboss/resteasy/cdi/ContextProducers.java#L45. The client itself is specifically done here https://github.com/resteasy/resteasy/blob/f8e03aabbca93fbaeecb21919343aa4484ae7ea1/resteasy-cdi/src/main/java/org/jboss/resteasy/cdi/ResteasyCdiExtension.java#L87-L116. The benefit injecting the client is that it's automatically closed for you.